c2rust-analyze/src/context.rs

```diff
@@ -26,27 +26,45 @@ bitflags! {
     /// That's why, for example [`UNIQUE`] is named as such,
     /// as opposed to something like `ALIASED` (a pointer capability),
     /// as removing [`UNIQUE`] (`&mut`) allows more values to be taken on (`&`).
+    ///
+    /// Furthermore, when "can" is used for a permission,
+    /// that does not mean that the pointer is always capable of, being [`READ`] for example.
+    /// That is, "can be read" means might be capable of being read,
+    /// or in other words, there are cases in which this pointer will be read.
```

Comment:
I don't think this is accurate.  If the pointer is `READ` then we will rewrite it into `&T` (or similar), which asserts to rustc that it is, in fact, valid and readable, regardless of whether it ever actually gets read/dereferenced.

This will change slightly when `NON_NULL` is added, since `READ` without `NON_NULL` will mean "either readable or null" (those being the valid states of a type like `Option<&T>`).

Reply:
I was basing this on the example you gave in https://github.com/immunant/c2rust/pull/818#discussion_r1099038252.

Reply:
This part has been removed now.  I think I was thinking ahead to when `NON_NULL` and `VALID` would be permissions, too, in which case this would apply (e.x. `READ` doesn't mean it's readable, as it could be null, only `READ | NON_NULL` does).